roles/kafka-mesos/tasks/scheduler.yml

```diff
@@ -19,7 +19,7 @@
     - kafka-mesos
 
 - name: Change Kafka scheduler configuration
-  when: kafka_app_status == '"{\"message\":\"An app with id [kafka-mesos-scheduler] already exists.\"}"'
+  when: kafka_app_status.stdout == '"{\"message\":\"An app with id [kafka-mesos-scheduler] already exists.\"}"'
```

Comment:
Pintless. Please make sure it works by running tests. If you don't know how to test it, I'll help you!